components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManagerTest.java

```diff
@@ -0,0 +1,228 @@
+
+        mockStatic(IdentityUtil.class);
+        when(IdentityUtil.extractDomainFromName(anyString())).thenReturn("testPrimaryDomain");
+
+        when(mockedUserStoreManager.getSecondaryUserStoreManager(anyString())).thenReturn(mockedUserStoreManager);
+        when(mockedUserStoreManager.isSCIMEnabled()).thenReturn(true);
+        when(mockedUserStoreManager.getRoleListOfUser(anyString())).thenReturn(userRoles);
+        mockStatic(AttributeMapper.class);
+        when(AttributeMapper.constructSCIMObjectFromAttributes(anyMap(), anyInt())).thenReturn(mockedUser);
+        when(mockedUserStoreManager.getRealmConfiguration()).thenReturn(mockedRealmConfig);
+        when(mockedRealmConfig.getEveryOneRoleName()).thenReturn("roleName");
+        when(mockedUserStoreManager.getTenantId()).thenReturn(1234567);
+        whenNew(GroupDAO.class).withAnyArguments().thenReturn(mockedGroupDAO);
+
+        SCIMUserManager scimUserManager = new SCIMUserManager(mockedUserStoreManager, mockedClaimManager);
+        assertNotNull(scimUserManager.getMe("testUserName", required));
+    }
+
+    @Test(dataProvider = "groupName")
+    public void testGetGroup(Object roleName, String userStoreDomain, Object expected) throws Exception {
+
+        whenNew(GroupDAO.class).withAnyArguments().thenReturn(mockedGroupDAO);
+        when(mockedGroupDAO.getGroupNameById(anyInt(), anyString())).thenReturn((String) roleName);
+        mockStatic(IdentityUtil.class);
+        when(IdentityUtil.extractDomainFromName(anyString())).thenReturn(userStoreDomain);
+        SCIMUserManager scimUserManager = new SCIMUserManager(mockedUserStoreManager, mockedClaimManager);
+        Group result = scimUserManager.getGroup("1234567", new HashMap<String, Boolean>());
+
+        String actual = null;
+        if (result != null) {
+            actual = result.getDisplayName();
+        }
+        assertEquals(actual, expected);
+    }
+
+    @DataProvider(name = "groupName")
+    public Object[][] groupName() throws Exception {
+
+        Group group = new Group();
+        group.setDisplayName("roleName");
+        return new Object[][]{
+                {null, "userStoreDomain", null},
+                {"roleName", null, "roleName"}
+        };
+    }
+
+    @Test(dataProvider = "getGroupException")
+    public void testGetGroupWithExceptions(String roleName, String userStoreDomain) throws Exception {
+
+        whenNew(GroupDAO.class).withAnyArguments().thenReturn(mockedGroupDAO);
+        when(mockedGroupDAO.getGroupNameById(anyInt(), anyString())).thenReturn((String) roleName);
+        mockStatic(IdentityUtil.class);
+        when(IdentityUtil.extractDomainFromName(anyString())).thenReturn(userStoreDomain);
+
+        SCIMUserManager scimUserManager = new SCIMUserManager(mockedUserStoreManager, mockedClaimManager);
+        try {
+            scimUserManager.getGroup("1234567", new HashMap<String, Boolean>());
```

Comment:
As I understand, you expect an exception to be thrown.So, just after this method call, you should add an assertFail();. If the flow executed as expected (ie. threw an exception), then the program won't reach the assertFail() line.

Reply:
I already tried this approach but the test method couldn't handle the exception when I assigned the specific exception as the expectedException.

Reply:
I'm not requesting you to set expectedException annotation parameter. :) I'm saying to add an #assertFail just after #getGroup method.